### PR TITLE
Xrender fixes

### PR DESCRIPTION
**Pull request by lokedhs**
Add the ability to pass `:NONE` as the `FORMAT` parameter for the triangle rendering functions.

Also added implementations of TriangleStrip and TriangleFan.

### CODE REVIEW

File: package.lisp [hunk 1]
```diff
@@ -177,4 +177,5 @@
     wm-size-hints-user-specified-position-p wm-size-hints-user-specified-size-p
     wm-size-hints-width wm-size-hints-width-inc wm-size-hints-win-gravity
     wm-size-hints-x wm-size-hints-y wm-zoom-hints write-bitmap-file
-    write-resources xatom))
+    write-resources xatom
+    #:render-set-picture-transform))
```
Comment (Harleqin) [inline package.lisp]: Why break local convention?  For one, this list uses “naked” symbols, so keep that.  If someone deems it worthwhile to change it, it should be done for the entire list in a separate commit.

But also: this list is sorted alphabetically, so the new symbol should also adhere to that (it should go to about line 133).  Be sure to keep within the given maximum line length.

**Comment by lokedhs** [comment]
@Harleqin thank you for your comments. I believe I have addressed your concerns in the following commit: 1af94444c1f0df5bdfb3124f7fd2a7766777c30f

**Comment by dkochmanski** [comment]
lgtm
